app/models/pillar.rb

```diff
@@ -65,7 +66,7 @@ def apply(pillars)
 
       Pillar.all_pillars.each do |key, pillar_key|
         # The following pillar keys can be blank, so ignore them if they are.
-        next if [:http_proxy, :https_proxy, :no_proxy].include?(key) && pillars[key].blank?
+        next if [:http_proxy, :https_proxy, :no_proxy, :insecure_registry].include?(key) && pillars[key].blank?
```

Comment:
If this list is getting larger, maybe it makes sense to have it in a constant or something like that on the top of the class. Or maybe on another method.